cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -181,9 +185,22 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			},
 		}
 
-		if vmSpec.Zone != "" {
-			zones := []string{vmSpec.Zone}
-			virtualMachine.Zones = &zones
+		// Set availability set if no failure domains are available
+		if len(s.Scope.FailureDomains()) == 0 {
+			if asName, ok := s.Scope.AvailabilitySet(); ok {
+				as, err := s.availabilitySetsClient.Get(ctx, s.Scope.ResourceGroup(), asName)
```

Comment:
no need to make a GET call for the ID here (API calls are expensive). The ID is deterministic, you can generate it from the name. See `SubnetID()`  in defaults.go and how it's used in networkinterfaces.go for an example.

Reply:
I'll take a look into it, Thanks!